### PR TITLE
feat(admin): admin users expansion

### PR DESCRIPTION
**Pull request by SotiriaSte**
**BREAKING CHANGE:**
Admin user (self) password change route is now a `PUT` operation
`[POST]` -> `[PUT]`: `/change-password`

**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [X ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [X] Yes
- [ ] No

**The PR fulfills these requirements:**

- [ ] It's submitted to the `main` branch
- [ ] When resolving a specific issue, it's referenced in the PR's description  (e.g. `fix #xxx`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature 

**Other information:**

### CODE REVIEW

File: packages/admin/src/routes/DisableTwoFa.route.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }),
+    async (params: ConduitRouteParameters) => {
+      const admin = params.context!.admin;
+      const context = params.context!;
+
+      if (isNil(context) || isNil(admin)) {
+        throw new GrpcError(status.UNAUTHENTICATED, 'Unauthorized');
+      }
+
+      await Admin.getInstance().findByIdAndUpdate(admin._id, {
```
Comment (kkopanidis) [inline packages/admin/src/routes/DisableTwoFa.route.ts:29]: should also delete the 2fa secret

File: packages/admin/src/routes/EnableTwoFa.route.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      const context = params.context!;
+
+      if (isNil(context) || isNil(admin)) {
+        throw new GrpcError(status.UNAUTHENTICATED, 'Unauthorized');
+      }
+      if (admin.hasTwoFA) {
+        return '2FA already enabled';
+      }
+
+      if (method === 'qrcode') {
```
Comment (kkopanidis) [inline packages/admin/src/routes/EnableTwoFa.route.ts:37]: there isn't any other method

File: packages/admin/src/routes/EnableTwoFa.route.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        const secret = generateSecret({
+          name: 'Conduit',
+          account: 'admin',
```
Comment (kkopanidis) [inline packages/admin/src/routes/EnableTwoFa.route.ts:40]: account should be equal to user's username

File: packages/admin/src/routes/GetAdminUsers.route.ts [hunk 4]
```diff
@@ -34,6 +34,18 @@ export function getAdminUsersRoute() {
             type: TYPE.String,
             required: true,
           },
+          hasTwoFA: {
+            type: TYPE.Boolean,
+            required: false,
+          },
+          twoFaMethod: {
```
Comment (kkopanidis) [inline packages/admin/src/routes/GetAdminUsers.route.ts:41]: remove twoFaMethod

File: packages/admin/src/routes/GetAdminUsers.route.ts [hunk 5]
```diff
@@ -34,6 +34,18 @@ export function getAdminUsersRoute() {
...
+            type: TYPE.String,
+            required: false,
+          },
+          isSuperAdmi: {
```
Comment (kkopanidis) [inline packages/admin/src/routes/GetAdminUsers.route.ts:45]: typo

**Review by kkopanidis — CHANGES_REQUESTED** [review]

File: packages/admin/src/routes/ChangeUsersPassword.route.ts [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  return new ConduitRoute(
+    {
+      path: '/change-user-password/:id',
+      action: ConduitRouteActions.UPDATE,
+      description: `Super Admin changes authenticated user's password.`,
+      urlParams: {
+        id: ConduitString.Required,
+      },
+      bodyParams: {
+        newPassword: ConduitString.Required,
```
[the remark this answers is not in the document]
Comment (kon14) [reply packages/admin/src/routes/ChangeUsersPassword.route.ts:23]: Let's skip this for now.

**Review by kon14 — CHANGES_REQUESTED** [review]

**Review by kon14 — APPROVED** [review]

**Review by kkopanidis — APPROVED** [review]

**Pull request merged**